launcher/java/JavaUtils.cpp

```diff
@@ -78,19 +87,22 @@ QProcessEnvironment CleanEnviroment()
             qDebug() << "Env: ignoring" << key << value;
             continue;
         }
-        // filter PolyMC-related things
-        if(key.startsWith("QT_"))
+
+        // These are used to strip the original variables
+        // If there is "LD_LIBRARY_PATH" and "LAUNCHER_LD_LIBRARY_PATH", we want to
+        // remove all values in "LAUNCHER_LD_LIBRARY_PATH" from "LD_LIBRARY_PATH"
+        if(key.startsWith("LAUNCHER_"))
         {
             qDebug() << "Env: ignoring" << key << value;
             continue;
         }
-#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
-        // Do not pass LD_* variables to java. They were intended for PolyMC
-        if(key.startsWith("LD_"))
+        if(stripped.contains(key))
         {
-            qDebug() << "Env: ignoring" << key << value;
-            continue;
+            QString newValue = stripVariableEntries(key, value, rawenv.value("LAUNCHER_" + key));
+
+            qDebug() << "Env: stripped" << key << value << "to" << newValue;
         }
+#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD)
```

Comment:
```suggestion
#if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
```

then this needs to change too x)